src/category_theory/functor_category.lean

```diff
@@ -51,6 +51,14 @@ lemma naturality_app {F G : C ⥤ (D ⥤ E)} (T : F ⟶ G) (Z : D) {X Y : C} (f
   ((F.map f).app Z) ≫ ((T.app Y).app Z) = ((T.app X).app Z) ≫ ((G.map f).app Z) :=
 congr_fun (congr_arg app (T.naturality f)) Z
 
+/-- A natural transformation is a monomorphism if each component is. -/
+lemma mono_app_of_mono (α : F ⟶ G) [∀ (X : C), mono (α.app X)] : mono α :=
```

Comment:
Do you want the `[..]`? I guess that typeclass inference will not be able to find those instances. Probably there would first need to be a `dsimp` that unfolds the `app` into something for which an instance exists.
And with the `[..]`, applying the lemma will fail if it can find the instance. So it might be better to use `(..)` and then call `apply_instance` in those cases where find the instance is actually possible.

Reply:
This is a fair point - I was mimicking what's done in `natural_isomorphism.lean`, where it has
```
def is_iso_of_is_iso_app (α : F ⟶ G) [∀ X : C, is_iso (α.app X)] : is_iso α :=
```
so I thought this version should be okay - I don't mind changing though

Reply:
Ooh well, if it's also done over there, let's try this. If it doesn't work, it's easy enough to change.